extensions/SharkPool/SoundCloud-API.js

```diff
@@ -0,0 +1,363 @@
+        return JSON.stringify(json.collection);
+      } catch { return "[]" }
+    }
+
+    async searchTracks(args) {
+      try {
+        let response;
+        args.QUERY = encodeURIComponent(args.QUERY);
+        if (!await Scratch.canFetch(`https://api-v2.soundcloud.com/search/tracks?q=${args.QUERY}&client_id=${clientID}&limit=${args.NUM2}&offset=${args.NUM1}&app_locale=en`)) return "";
+        // eslint-disable-next-line
+        response = await fetch(`${proxy}https://api-v2.soundcloud.com/search/tracks?q=${args.QUERY}&client_id=${clientID}&limit=${args.NUM2}&offset=${args.NUM1}&app_locale=en`);
+        if (!response.ok) return "Error: 404";
+        let json = await response.json();
+        return JSON.stringify(json.collection);
+      } catch { return "[]" }
+    }
+
+    async getArtistAtt(args) {
+      try {
+        let response;
+        if (!await Scratch.canFetch(`https://api-v2.soundcloud.com/users/${args.ID}?client_id=${clientID}&app_locale=en`)) return "";
+        // eslint-disable-next-line
+        response = await fetch(`${proxy}https://api-v2.soundcloud.com/users/${args.ID}?client_id=${clientID}&app_locale=en`);
+        if (!response.ok) return "Error: 404";
+        let json = await response.json();
+        return json[args.THING] ?? "";
+      } catch { return "" }
+    }
+
+    async getFollowers(args) {
+      try {
+        let response;
+        if (!await Scratch.canFetch(`https://api-v2.soundcloud.com/users/${args.ID}/followers?client_id=${clientID}&limit=${args.NUM2}&offset=${args.NUM1}&app_locale=en`)) return "";
+        // eslint-disable-next-line
+        response = await fetch(`${proxy}https://api-v2.soundcloud.com/users/${args.ID}/followers?client_id=${clientID}&limit=${args.NUM2}&offset=${args.NUM1}&app_locale=en`);
+        if (!response.ok) return "Error: 404";
+        let json = await response.json();
+        return JSON.stringify(json.collection);
+      } catch { return "[]" }
+    }
+
+    async getTracks(args) {
+      try {
+        let response;
+        if (!await Scratch.canFetch(`https://api-v2.soundcloud.com/users/${args.ID}/tracks?representation=&client_id=${clientID}&limit=${args.NUM2}&offset=${args.NUM1}&app_locale=en`)) return "";
+        // eslint-disable-next-line
+        response = await fetch(`${proxy}https://api-v2.soundcloud.com/users/${args.ID}/tracks?representation=&client_id=${clientID}&limit=${args.NUM2}&offset=${args.NUM1}&app_locale=en`);
+        if (!response.ok) return "Error: 404";
+        let json = await response.json();
+        return JSON.stringify(json.collection);
+      } catch { return "[]" }
+    }
+
+    async searchArtists(args) {
+      try {
+        let response;
+        args.QUERY = encodeURIComponent(args.QUERY);
+        if (!await Scratch.canFetch(`https://api-v2.soundcloud.com/search/users?q=${args.QUERY}&client_id=${clientID}&limit=${args.NUM2}&offset=${args.NUM1}&app_locale=en`)) return "";
+        // eslint-disable-next-line
```

Comment:
Use `Scratch.fetch` or write a comment why you can't. It's important to know for maintenance.